extensions/graphql/README.md

```diff
@@ -0,0 +1,179 @@
+# @loopback/graphql
+
+This module provides integration with [GraphQL](https://graphql.org/) using
+[type-graphql](https://typegraphql.com/).
+
+![type-graphql](type-graphql.png)
+
+## Stability: ⚠️Experimental⚠️
+
+> Experimental packages provide early access to advanced or experimental
+> functionality to get community feedback. Such modules are published to npm
+> using `0.x.y` versions. Their APIs and functionality may be subject to
+> breaking changes in future releases.
+
+## Basic Use
+
+```ts
+export class MyApplication extends BootMixin(RestApplication) {
```

Comment:
It is strange to see **GraphQL**Component with **Rest**Application. If `RestApplication` can be used this way, maybe it should not be named `RestApplication`? Maybe just `WebApplication`?

The application type is then determined by whether you do:

1. `this.component(RestExplorerComponent);`
or
2. `this.component(GraphQLComponent);`
or
3. `this.component(GrpcComponent);`
etc.

I think this observation is an indication that `RestApplication` should be made more generic, and the REST-specific parts should be provided by the Component.

@bajtos @raymondfeng what do you think?

Reply:
REST/GraphQL both use `http/https` as the underlying protocol. `gRPC` uses `http/2`. I don't see a urgent need to rename `RestApplication` to `HttpApplication`.

Reply:
Agreed it's not urgent. However, it is not just renaming to `HttpApplication`, we will need to extract the REST-specific parts from `RestApplication` and enable REST functionality the way we enable GraphQL or other functionalities. In the current state, `RestComponent` is loaded by default in `RestApplication `. 

Or will this work?

```ts
class GraphQlApplication extends Application implements HttpServerLike
...
MyApplication extends BootMixin(GraphQlApplication) 
```

Nah, it would be repetitive code or create more files. The better approach would be to rename `RestApplication`, make it more generic, and extract the REST bits out of it.

@bajtos thoughts?

Reply:
If GraphQL support is to be implemented as an extension, REST should be no different.

Reply:
I feel we have bigger fish to fry, I am fine to use _GraphQLComponent with RestApplication_ for initial iterations. Let's build a better understanding of what our users need and how they are going to use GraphQL before we jump on refactorings.